### ISSUE TITLE
Game ending priority

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#692]
Original [issue 692](https://code.google.com/p/pychess/issues/detail?id=692) reported by [stevemulligan](https://code.google.com/u/102948650107978762260/) 2011-10-10

It looks like getResults (the check for game endings) has an issue with ending priorities.  It looks like it checks for repetition and 50 move rule draws before checkmate.  It doesn't happen very often but I was trying to use pyChess to generate some test cases and this caused some problems.

Utils/logic.py  around line 40.

I made the following modifications so I could get going on my project but I removed the LOSERSCHESS variant (Not familiar with it, sorry)

def getStatus (board):
    lboard = board.board
    board_clone = lboard.clone()
    hasMove = False
    for move in lmovegen.genAllMoves (board_clone):
        board_clone.applyMove(move)
        if board_clone.opIsChecked():
            board_clone.popMove()
            continue
        board_clone.popMove()
        hasMove = True
        break

```
if not hasMove:
    if lboard.isChecked():
        if board.color == WHITE:
            status = BLACKWON
        else:
            status = WHITEWON
        return status, WON\_MATE
    else:
        return DRAW, DRAW\_STALEMATE

if ldraw.repetitionCount (lboard) >= 3:
    return DRAW, DRAW\_REPITITION

if ldraw.testFifty (lboard):
    return DRAW, DRAW\_50MOVES

return RUNNING, UNKNOWN\_REASON
```

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=692#c1) posted by [gbtami](https://code.google.com/u/gbtami/) 2011-10-14

Good catch, thx!
- changed status to: Accepted
- added label: Defect

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=692#c2) posted by [gbtami](https://code.google.com/u/gbtami/) 2011-10-14

This issue was closed by revision 2f03c3d00c58.
- changed status to: Fixed
